architecture.md

```diff
@@ -0,0 +1,44 @@
+# Architecture
+
+This document describes the high-level architecture of the Gleam transpiler.
+
+## Overview
+
+On the highest level, a `gleam` binary receives Gleam source code as input
+and produces one of its available backend formats:
+
+- Erlang source code (not Erlang Abstract Format, as e.g. Elixir does)
+- Javascript source code
+
+There is currently no formal specification of Gleam source,
+apart from the programmer-oriented [Language tour](https://gleam.run/book/tour/index.html).
```

Comment:
Can also note the integration tests in `test/language`. I would like to cover most the language runtime behaviour there.